File: login/src/main/java/com/example/login/damain/user/entity/repository/UserRepository.java
package com.example.login.damain.user.entity.repository;

import com.example.login.damain.user.entity.User;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface UserRepository extends JpaRepository<User, Integer> {
    Optional<User> findByEmail(String email);
    boolean existsByEmail(String email);
}

File: practice-kotlin/src/main/kotlin/com/example/practicekotlin/domain/user/domain/repository/UserRepository.kt
package com.example.practicekotlin.domain.user.domain.repository

import com.example.practicekotlin.domain.user.domain.User
import org.springframework.data.repository.CrudRepository

interface UserRepository: CrudRepository<User, Long> {

    fun findByAccountId(accountId: String): User?

    fun existsByAccountId(accountId: String): Boolean

}